From 1f95be543bb707d5eb0bc1c6f35f0fe37969383f Mon Sep 17 00:00:00 2001
From: Hu Kejun <william.hu@rock-chips.com>
Date: Mon, 18 Mar 2019 18:07:22 +0800
Subject: [PATCH] media: spi: RK1608: support no sensor connect to preisp

To support preisp post-processing and camera hal1,
let driver probe ok when no sensor device is connected.

Change-Id: I1830420f448b47a0bd327ee3950da0eb7af8d3fb
Signed-off-by: Hu Kejun <william.hu@rock-chips.com>
---
 drivers/media/spi/rk1608.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/media/spi/rk1608.c b/drivers/media/spi/rk1608.c
index 453ae923f7d9..99fc5e9cc254 100644
--- a/drivers/media/spi/rk1608.c
+++ b/drivers/media/spi/rk1608.c
@@ -1518,10 +1518,8 @@ static int rk1608_probe(struct spi_device *spi)
 	}
 
 	ret = rk1608_get_remote_node_dev(rk1608);
-	if (ret) {
-		dev_err(rk1608->dev, "Get remote node dev err %x\n", ret);
-		return ret;
-	}
+	if (ret)
+		dev_info(rk1608->dev, "remote node dev is NULL\n");
 
 	rk1608->sensor_cnt = 0;
 	mutex_init(&rk1608->sensor_lock);
-- 
2.35.3

